src/Core/Checkout/Cart/Rule/LineItemCustomFieldRule.php

```diff
@@ -109,4 +109,48 @@ private function isCustomFieldValid(LineItem $lineItem): bool
                 throw new UnsupportedOperatorException($this->operator, self::class);
         }
     }
+
+    /**
+     * @return Constraint[]
+     */
+    private function getRenderedFieldValueConstraints(): array
+    {
+        $constraints = [];
+
+        if (!\is_array($this->renderedField) || !\array_key_exists('type', $this->renderedField)) {
+            return [new NotBlank()];
+        }
+
+        if ($this->renderedField['type'] !== CustomFieldTypes::BOOL) {
+            $constraints[] = new NotBlank();
+        }
+
+        return $constraints;
+    }
+
+    private function getValue(array $customFields, array $renderedField)
+    {
+        $value = null;
+
+        if (in_array($renderedField['type'], [CustomFieldTypes::BOOL, CustomFieldTypes::SWITCH], true)) {
+            if (!empty($customFields) && \array_key_exists($this->renderedField['name'], $customFields)) {
+                $value = $customFields[$renderedField['name']];
```

Comment:
is it possible to use early returns here, instead of assigning the $value to the variable?

Reply:
Sure, no problem. Changed accordingly.